### ISSUE TITLE
Application is creating a directory named "~" for storing data

### ISSUE THREAD

**Issue by Kreeblah** [tpill90/steam-lancache-prefill#381]
**Describe the bug**
This is with version 2.7.0 on macOS 14.6.1 on an M1 Max Macbook Pro.

I noticed after filing https://github.com/tpill90/steam-lancache-prefill/issues/380 that I had a directory named "~" alongside the `SteamPrefill` binary.  This seemed odd, so I looked in there and found a directory structure that looks like `Library/Caches/SteamPrefill/v1/`.  So, apparently that's not being appropriately replaced with my home directory.

**Additional Details**
If I delete this directory, it is recreated the next time I attempt to run `SteamPrefill`.

**Screenshots**

**Logs**
Please attach the `app.log` found in the application's directory to this issue.

**Comment by tpill90** [comment]
I don't have a Mac to test this with, but could you please let me know what the value of the environment variable "$PATH" is on your machine?  I'm wondering if that should be what I'm looking at instead.

**Comment by Kreeblah** [comment]
I suspect you probably want to look at `$HOME`.  This is what I have for that:

```
fsh " echo $HOME
/Users/myuser
```

If you're still curious about `$PATH`, though, here's what mine looks like:

```
fsh " echo $PATH
/opt/homebrew/bin /opt/homebrew/sbin /opt/local/sbin /opt/local/bin /Users/myuser/.cargo/bin /Applications/ShellHistory.app/Contents/Helpers /Users/myuser/.local/bin /Users/myuser/.fig/bin /usr/local/bin /System/Cryptexes/App/usr/bin /usr/bin /bin /usr/sbin /sbin /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin /var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin /opt/X11/bin /Applications/Pololu USB AVR Programmer v2.app/Contents/MacOS /Library/Apple/usr/bin /Applications/Little Snitch.app/Contents/Components /Applications/Server.app/Contents/ServerRoot/usr/bin /Applications/Server.app/Contents/ServerRoot/usr/sbin /Applications/VMware Fusion.app/Contents/Public /usr/local/share/dotnet ~/.dotnet/tools /Library/TeX/texbin /Applications/Wireshark.app/Contents/MacOS
```

**Comment by tpill90** [comment]
You're right, I was looking for `$HOME`.  Not sure what my brain was thinking.

I'm not familiar with where MacOS keeps temporary files, but would `/Users/myuser/Library/Caches` be the right place to be sticking SteamPrefill's temp data?

**Comment by Kreeblah** [comment]
I guess that depends.  If we're talking temporary for the lifetime of the process, then `$TMPDIR` is probably going to be what you want.  Each process automatically gets a randomly-named directory created for it in `/var/folders/nk/` that's available in that environment variable, and it's also automatically cleaned up when the process exits.  For example, the value of it for a terminal session I have open right now looks like this:

```
$ echo $TMPDIR
/var/folders/nk/s955071d24d2lx0v7_f0jd140000gn/T/
```

Anything I were to put in there would be automatically removed when my terminal session ends.  Each process that I run in the terminal gets its own directory as well.

If it needs to persist through runs, then, yeah, something like `$HOME/Library/Caches/steam-lancache-prefill` could be a good place for them.  That's typically used for files that are persisted for performance reasons, but wouldn't negatively impact things (other than whatever time it takes to regenerate them) if they're removed (so . . . caches).

If it's related to user-generated data, then `$HOME/Library/Application Support/steam-lancache-prefill` would be the place for that.

There's also a `$HOME/Library/Preferences` directory which is used for user preferences, but it's only supposed to be written to via system calls, and not by writing files directly.  I see a lot of applications use `$HOME/Library/Application Support` instead when they aren't written to use the macOS native APIs.

https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/MacOSXDirectories/MacOSXDirectories.html#//apple_ref/doc/uid/TP40010672-CH10-SW1 has more info about canonical directory locations, if you're curious.

**Comment by tpill90** [comment]
Yeah this data will be persisted through runs, just saving manifests so that they don't need to be redownloaded again.  Sounds like `$HOME/Library/Caches` is where I'll stick these manifests.

**Comment by tpill90** [comment]
Released in [v3.1.0](https://github.com/tpill90/steam-lancache-prefill/releases/tag/v3.1.0)
